### PR TITLE
Reduce overhead for point logging

### PR DESCRIPTION
**Pull request by basilisk487**

### CODE REVIEW

File: proxy/src/main/java/com/wavefront/agent/PointHandlerImpl.java [hunk 1]
```diff
@@ -58,11 +71,17 @@ public PointHandlerImpl(final String handle,
     this.blockedPointsPerBatch = blockedPointsPerBatch;
     this.prefix = prefix;
     String logPointsProperty = System.getProperty("wavefront.proxy.logpoints");
-    this.logPoints = logPointsProperty != null && logPointsProperty.equalsIgnoreCase("true");
+    this.logPointsFlag = logPointsProperty != null && logPointsProperty.equalsIgnoreCase("true");
 
     this.receivedPointLag = Metrics.newHistogram(new MetricName("points." + handle + ".received", "", "lag"));
 
     this.sendDataTasks = sendDataTasks;
+    this.pointHandlerExecutor =  Executors.newScheduledThreadPool(1,
```
Comment (panghy) [inline proxy/src/main/java/com/wavefront/agent/PointHandlerImpl.java]: this seems a bit over-complicated (to take a thread to do this). perhaps just add a volatile long and check the loggable status if it's stale instead?

**Comment by basilisk487** [comment]
@panghy Done

File: proxy/src/main/java/com/wavefront/agent/PointHandlerImpl.java [hunk 2]
```diff
@@ -33,15 +33,11 @@
...
-  /**
-   * Executor for support tasks (refreshing logger level)
-   */
-  private final ScheduledExecutorService pointHandlerExecutor;
-
   private final Histogram receivedPointLag;
   private final String validationLevel;
   private final String handle;
   private boolean logPoints = false;
+  private long logPointsUpdatedMillis = 0L;
```
Comment (panghy) [inline proxy/src/main/java/com/wavefront/agent/PointHandlerImpl.java]: volatile to be absolutely correct. setting the value of a long might not be atomic

Comment (panghy) [reply proxy/src/main/java/com/wavefront/agent/PointHandlerImpl.java]: oh you did it later

**Pull request merged**